pytest_datadir/plugin.py

```diff
@@ -1,38 +1,22 @@
 import os
+import pathlib
 import shutil
 
 import pytest
 
 
-class TestDataDir(object):
-
-    def __init__(self, global_dir, module_dir, tmpdir):
-        self.global_dir = global_dir
-        self.module_dir = module_dir
-        self.tmpdir = tmpdir.strpath
-        assert os.path.isdir(self.module_dir) or os.path.isdir(self.global_dir),\
-            'neither {} or {} are valid data directories'.format(self.global_dir, self.module_dir)
+@pytest.fixture
+def global_datadir(request):
```

Comment:
Also it is not really "global" because it only looks for a "data" folder on the same directory as the test file requesting it.